lib/ClangImporter/ClangImporter.cpp

```diff
@@ -7539,11 +7543,9 @@ ValueDecl *ClangImporter::Implementation::importBaseMemberDecl(
   // Make sure we don't clone the decl again for this class, as that would
   // result in multiple definitions of the same symbol.
   std::pair<ValueDecl *, DeclContext *> key = {decl, newContext};
-  auto known = clonedBaseMembers.find(key);
-  if (known == clonedBaseMembers.end()) {
-    ValueDecl *cloned = cloneBaseMemberDecl(decl, newContext);
-    known = clonedBaseMembers.insert({key, cloned}).first;
-  }
+  auto [known, inserted] = clonedBaseMembers.try_emplace(key, nullptr);
+  if (inserted)
+    known->getSecond() = cloneBaseMemberDecl(decl, newContext);
```

Comment:
This looks somewhat risky – what if `cloneBaseMemberDecl` modifies `clonedBaseMembers` and that invalidates the iterator?

Reply:
I am worried about this too since `cloneBaseMembers` is a fairly chunky piece of code and it is difficult to enforce or locally verify the invariant that `clonedBaseMembers` is not being modified in that function.

I don't think `cloneBaseMembers` should ever _need_ to touch the cache, but I am very slightly leaning toward reverting this change just to be safe.

Reply:
Reverted in the newest iteration.